### DISCUSSION TITLE
Can't get whisper to run on GPU (windows 10)

### DISCUSSION THREAD

**Discussion by an303042** [openai/whisper#672] [Q&A] [answered]
Hello,
Been trying to run Whisper locally, but keep having issues.
While it isn't really a whisper issue, but probably its dependencies, I figured I'd ask here.
So I've installed cuda toolkit, I have python installed and in the PATH, I am able to run whisper on cpu, but when trying to add --device cuda, i get the following: "RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU."

However, If I run python, and import torch I get - 
>>> print(torch.__version__)
1.13.0+cu117
>>> print(torch.cuda.is_available())
True

So I don't know what to do.
my machine is windows 10 with 2070 Super.
Any help would be greatly appreciated.

**Comment by ghost** [comment]
If I'm not totally wrong, you must install CUDA 11.7 with [cudnn Version 8](https://developer.nvidia.com/cudnn). At least I have done this and it works. The Torch packet is tailored for CUDA 11.7 but still requires the CUDA 11.7 runtime.

**Comment by jongwook** [comment, marked as answer]
It's also possible that where your `pip` installs packages to is different from your `python` command is pointing to. Can you try:

```bash
python -m whisper audio.wav
```

instead of `whisper audio.wav`?

**Reply by an303042** [reply]
Thank you! Can't believe that's all it took

**Reply by Holden141** [reply]
came from the future to say thank you!

**Comment by Baenwort** [comment]
Has anyone gotten pytourch to use OpenCL instead of Cuda with whisper?

**Reply by err4nte** [reply]
I've learnt that you can use ROCm 5.2 in linux and that there is something called HIP that translate cuda into amd or something like that.
Still haven't been able to make it work :(
